addon/components/zero-clipboard.js

```diff
@@ -2,21 +2,29 @@ import Ember from 'ember';
 /* global ZeroClipboard */
 
 export default Ember.Component.extend({
-	attributeBindings: ['title', 'data-clipboard-text', 'data-clipboard-target'],
-	title: 'Copy to clipboard',
-	didInsertElement: function () {
-	  var client = new ZeroClipboard(this.get('element'));
+  attributeBindings: ['title', 'data-clipboard-text', 'data-clipboard-target'],
+  title: 'Copy to clipboard',
+  zeroClipboardEvents: ['ready', 'beforeCopy', 'copy', 'afterCopy', 'destroy', 'error'],
+  didInsertElement: function () {
+    var client = new ZeroClipboard(this.get('element'));
+    var _this = this;
+    //bind Zero Clipboard events to ember events
+    this.get('zeroClipboardEvents').forEach(function(action){
+      client.on(action, Ember.run.bind(this, function(event) {
+        try {
+					this.send(action, event);
+				}
+				catch(error) {
+					Ember.Logger.debug(error.message);
+				}
+      }));
+    }, _this);
+  },
+  "data-clipboard-text": function(){
+    return this.get('text');
+  }.property('text'),
 
-		//bind aftercopy to an ember event
-	  client.on("aftercopy", Ember.run.bind(this, function(event) {
-			this.send('afterCopy', event);
-	  }));
-	},
-	"data-clipboard-text": function(){
-		return this.get('text');
-	}.property('text'),
-
-	"data-clipboard-target": function(){
-		return this.get('cbTarget');
-	}.property('cbTarget')
+  "data-clipboard-target": function(){
```

Comment:
Sure, and I can also fix the awesome indentation that my editor seems bent on enforcing.